src/batch/src/executor/join/lookup_join.rs

```diff
@@ -88,62 +125,56 @@ impl<C: BatchTaskContext> ProbeSideSource<C> {
             ).into());
         }
 
-        let eq_conds: Vec<_> = self
-            .build_side_idxs
-            .iter()
-            .map(|&idx| {
-                cur_row
-                    .value_at(idx)
-                    .to_owned_datum()
-                    .unwrap()
-                    .to_protobuf()
-            })
-            .collect();
-
-        let scan_ranges = if eq_conds.is_empty() {
+        let scan_ranges = if scan_range.eq_conds.is_empty() {
```

Comment:
In what case it's empty?

Reply:
I don't think it will ever be empty actually, this check can be removed.